compiler-plugin/src/main/kotlin/com/google/devtools/ksp/KotlinSymbolProcessingExtension.kt

```diff
@@ -204,6 +215,21 @@ abstract class AbstractKotlinSymbolProcessingExtension(val options: KspOptions,
     private fun KSPLogger.hasError(): Boolean {
         return (this as MessageCollectorBasedKSPLogger).recordedEvents.any { it.severity == CompilerMessageSeverity.ERROR || it.severity == CompilerMessageSeverity.EXCEPTION }
     }
+
+    private fun handleException(call: () -> Unit) {
+        try {
+            call()
+        } catch (e: Exception) {
+            // Throws KSP exceptions
+            if (e.stackTrace.first().className.startsWith("com.google.devtools.ksp")) {
```

Comment:
Can we avoid the hard coded string, or define the constant in a testable (by reflection) function / variable.